### PR TITLE
Recipes

### PR DESCRIPTION
**Pull request by pelikhan**
Essencially, allows to start a tutorial within an existing program. In Aracde, this means that "game concepts" can be followed while making a game.

- [x] slurp list of recipes from the /recipes markdown page
- [x] adding experiment
- [x] don't leave editor when finishing tutorial
- [x] add "tutorials" button in toolbox above advance
- [x] filter tutorials based on editor type

recipe

Use https://github.com/microsoft/pxt-arcade/tree/ututorials to test locally in arcade.

### CODE REVIEW

File: webapp/src/app.tsx [hunk 1]
```diff
@@ -2719,29 +2725,32 @@ export class ProjectView
...
         let reportId: string = undefined;
-        let tutorialmd: string;
         let title: string;
         let autoChooseBoard: boolean = true;
         let dependencies: pxt.Map<string> = {};
         let features: string[] = undefined;
         let p: Promise<string>;
+
+        // make sure we are in the editor
+        recipe = recipe && !!this.state.header;
```
Comment (darzu) [inline webapp/src/app.tsx]: Can a regular tutorial run when not in the editor?

File: webapp/src/app.tsx [hunk 2]
```diff
@@ -3299,6 +3328,11 @@ function handleHash(hash: { cmd: string; arg: string }, loading: boolean): boole
             editor.startTutorial(hash.arg);
             pxt.BrowserUtils.changeHash("");
             return true;
+        case "recipe": // shortcut to a tutorial. eg: #tutorial:tutorials/getting-started
+            pxt.tickEvent("hash.tutorial")
```
Comment (darzu) [inline webapp/src/app.tsx]: new ticket event? hash.recipe

Comment (pelikhan) [reply webapp/src/app.tsx]: good catch

File: webapp/src/app.tsx (hunk 1 above)
Comment (pelikhan) [reply webapp/src/app.tsx]: It can be triggered by a hash change

**Comment by pelikhan** [comment]
@shakao could you check that i merged your changes properly?

**Comment by pelikhan** [comment]
This is ready for review. Updates from UI meeting integrated.

**Comment by shakao** [comment]
looks good to me!

**Pull request merged**